apps/root/src/state/balances/actions.ts

```diff
@@ -99,7 +100,21 @@ export const fetchInitialBalances = createAppAsyncThunk<BalancesState['balances'
       chainIds,
     });
 
-    for (const [walletAddress, chainBalances] of Object.entries(accountBalancesResponse.balances)) {
+    // Merging api balances with customToken balances
+    const mergedBalances = cloneDeep(accountBalancesResponse.balances);
+    const customTokens = getState().tokenLists.customTokens.tokens;
+
+    for (const walletAddress of wallets) {
+      const customTokensBalances = await sdkService.getMultipleBalances(customTokens, walletAddress);
+
+      for (const [chainId, balanceList] of Object.entries(customTokensBalances)) {
+        for (const [tokenAddress, balance] of Object.entries(balanceList)) {
+          mergedBalances[walletAddress][Number(chainId)][tokenAddress] = balance.toString();
```

Comment:
shouldn't we check that '[walletAddress][chainId]' existsfirst?

Reply:
True! Thanks